### DISCUSSION TITLE
keyball39　片側キーボードしか動かない、トラックボールが作動しない、タクトスイッチを押すと何が起きるのか

### DISCUSSION THREAD

**Discussion by jun3366** [Yowkees/keyball#161] [Q&A]
以前メールにて不明点について問い合わせしていましたが、他の方も閲覧できるこのサイトに質問させていただき、
今後使われる方の参考になればと思い、こちらに質問致します。

keyball39について下記内容質問致します。

①タクトスイッチを押すと何が起きているのかご教授いただきたいです。
以前は左手側にPCからのケーブルを繋いでいて両側で入力できていました。
両側入力できていた時にトラックボールが作動しなかったため、タクトスイッチを1回押すと片側入力しかできなくなってしまいました。（のちにトラックボールの再はんだを行いトラックボールは動くようになったが、右側タクトスイッチを押すとまた動かなくなった）
タクトスイッチは何のためについているのか、1回押すのと2回素早く押すのとでどういうことが起きているのか知りたいです。

②片側キーボードしか動かない
TRSケーブルを左右に差しているが、PCとケーブルを指している側しか作動しません。
※右側をPCに接続すると左側のOLEDに「keyball」と表示されているので通電はしていると思います。
両側同時に作動させる解決方法お聞きしたいです。

③トラックボールが動かない
右手側にトラックボールは位置しています。
右手側、左手側にPCからのケーブルを指してもトラックボールが動きません。
（②でも記載したとおり右手側にPCからのケーブルを繋いだらキーボード入力はできます。）

長文で恐縮ですが、ご回答よろしくお願いします。

**Comment by Yowkees** [comment]
ご質問ありがとうございます。

下記にご回答させていただきます。

①タクトスイッチはリセットスイッチです。
リセットスイッチを押すとProMicroが書込可能モード(DFUモード)に入り、キーボードとしての動作は停止します。
リセットスイッチを使用するのはファームウェアを書き換えるときだけで、通常使用時には使用しません。

②左右の通信がうまくできていないようですので、TRSコネクタの半田付けを見直してください。
きちんと基板の銀色パッドまで半田が届いていないと導通していないことが多いです。
半田が山型になっているでしょうか？

改善しない場合にはProMicro本体の半田付けも見直してください。

③①にて再はんだで改善したとありますが、別の問題でしょうか？おそらく②が改善されれば直りそうですのでまず②の作業をまず行ってください。

お手数ですがご対応よろしくお願いします。

**Reply by jun3366** [reply]
ご回答ありがとうございます。
①タクトスイッチの使い方理解しました。

②はんだ付けを再度行ったら両側で動くようになりました。ありがとうございます。

③トラックボールが動かないです。
ボールケースを前後左右上下に動かしたら稀に動くことはありますが、ボールを転がしたスムーズな動きではなく、
マウスポインターが暴れて制御できないような挙動を見せます。
トラックボール用基板の接触不良が問題でしょうか？
上記情報で考えられる対応方法などございましたらご教授お願い致します。

**Reply by jun3366** [reply]
③のトラックボールが動かない件ですが，ボール用基板を再はんだを実施したら，ボールが反応しました。
ただ，トラックボールを動かさず，マウスポインタ―が少しずつ上に勝手に上がって行ってしまったり，
その場に留まってくれない症状が出ています。
解決策ご教授お願いいたします。
